File: app/src/main/java/com/bird/launcher/session/SessionContract.java
package com.bird.launcher.session;

import android.net.Uri;

public final class SessionContract {
    public static final String AUTHORITY = "com.bird.launcher.provider";
    public static final String CONTENT_URI_BASE = "content://" + AUTHORITY;

    private SessionContract() {
    }

    public static final class Session {
        public static final String PATH = "session";
        public static final String CONTENT_URI_STRING = CONTENT_URI_BASE + "/" + PATH;
        public static final Uri CONTENT_URI = Uri.parse(CONTENT_URI_STRING);
        public static final String CONTENT_TYPE = "vnd.android.cursor.item/vnd.com.bird.launcher.session";
        public static final String COLUMN_USERNAME = "username";
        public static final String COLUMN_LOGGED_IN = "logged_in";

        private Session() {
        }
    }
}

File: app/src/main/java/com/bird/StarrySkyTeaHouse/session/SessionContract.kt
package com.bird.StarrySkyTeaHouse.session

import android.net.Uri

object SessionContract {
    const val AUTHORITY = "com.bird.StarrySkyTeaHouse.provider"
    const val READ_PERMISSION = "com.bird.StarrySkyTeaHouse.permission.READ_SESSION"
    const val CONTENT_URI_BASE = "content://$AUTHORITY"

    object Session {
        const val PATH = "session"
        const val CONTENT_URI_STRING = "$CONTENT_URI_BASE/$PATH"
        val CONTENT_URI: Uri = Uri.parse(CONTENT_URI_STRING)
        const val CONTENT_TYPE = "vnd.android.cursor.item/vnd.com.bird.StarrySkyTeaHouse.session"
        const val COLUMN_USERNAME = "username"
        const val COLUMN_LOGGED_IN = "logged_in"
    }
}
